src/libos/src/signal/do_sigtimedwait.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::Weak;
+use std::time::Duration;
+
+use super::{siginfo_t, SigNum, SigSet, Signal};
+use crate::events::{Observer, Waiter, WaiterQueueObserver};
+use crate::prelude::*;
+use crate::process::{ProcessRef, TermStatus, ThreadRef};
+
+pub fn do_sigtimedwait(interest: SigSet, timeout: Option<&Duration>) -> Result<siginfo_t> {
+    debug!(
+        "do_rt_sigtimedwait: interest: {:?}, timeout: {:?}",
+        interest, timeout,
+    );
+
+    let thread = current!();
+    let process = thread.process().clone();
+
+    // Interesting, blocked signals
+    let interest = {
+        let blocked = thread.sig_mask().read().unwrap();
+        *blocked & interest
+    };
+
+    let signal = match timeout {
```

Comment:
This [link](https://man7.org/linux/man-pages/man2/sigtimedwait.2.html) says: Attempts to wait for SIGKILL and SIGSTOP are silently ignored.

Reply:
Yes, that's the right behavior. But the code actually is correct. This is because `blocked`, which is returned from `thread.sig_mask()`, never contains `SIGKILL` and `SIGSTOP`. If so, this would already be a bug of `sigprocmask` syscall.

Reply:
Yes, you are right about the expected behaviors for `SIGKILL` and `SIGSTOP`. But, the code is correct. The reason is that `blocked`, which is initialized as the signal mask of the current thread, cannot contain `SIGKILL` or `SIGSTOP`. So the result of `*blocked & interest` does not contain `SIGKILL` or `SIGSTOP`, either.